app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < Sequel::Model
+  plugin :validation_helpers
+  raise_on_save_failure = false
+
+  def validate
+    super
+    validates_presence [:fname, :lname, :email, :username]
+    validates_min_length 3, [:fname, :lname, :email, :username]
+    validates_max_length 100, [:fname, :lname, :email, :username]
+    validates_format /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, :email
+    # validates_unique(:email, :username, [:fname, :lname]) # having trouble using this when updating info
```

Comment:
I commented validates_unique because I'm having trouble using this when updating user info. It doesn't allow the user to keep their original info if they want to update it.